src/inventory.cpp

```diff
@@ -867,15 +881,32 @@ void Inventory::deSerialize(std::istream &is)
 			std::getline(iss, listname, ' ');
 			iss>>listsize;
 
-			InventoryList *list = new InventoryList(listname, listsize, m_itemdef);
+			InventoryList *list = getList(listname);
+			bool create_new = !list;
+			if (create_new)
+				list = new InventoryList(listname, listsize, m_itemdef);
+			else
+				list->setSize(listsize);
 			list->deSerialize(is);
 
-			m_lists.push_back(list);
-		}
-		else
-		{
-			throw SerializationError("invalid inventory specifier: " + name);
+			new_lists.push_back(list);
+			if (create_new)
+				m_lists.push_back(list);
+
+		} else if (name == "KeepList") {
+			// Incrementally sent list
+			std::string listname;
+			std::getline(iss, listname, ' ');
+
+			InventoryList *list = getList(listname);
+			if (list)
+				new_lists.push_back(list);
+			else
+				errorstream << "Inventory::deSerialize(): Tried to keep list '" <<
+					listname << "' which is non-existent." << std::endl;
```

Comment:
braces

Reply:
The style rules depend on statements, not lines.

> Do not unnecessarily use braces where a single statement will do.
>
>	if (condition)
>		action();

https://www.kernel.org/doc/Documentation/process/coding-style.rst
Do you insist on adding them anyway?

Reply:
I don't think I've seen **multi-line** if/else statements without braces anywhere else, so it'd still be nice if you could add them.